### PR TITLE
feat(ui): render page using sign-in exp settings

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Render page using sign-in exp settings

This PR includes following updates:
1. Fix the CN phrase sign-in methods link content bug
2. Make confirm modal index higher than social popUp
3. Sign-In input fields styling updates
4. Update PrimarySocialSignInr and SecondarySocialSignIn to load the social connectors from the global sign-in experience settings provided by `useSocial` hook 
5. update useSocial hook to load the social connector's settings from page context and merge with the native supported ones in any. 
6. Add the `SocialPopUp` modal to the `SecondarySocialSignIn` container. Close the modal when `invokeSocialSignIn` request returns. 
7.  Create the PrimarySection & SectiondarySection render components to provide the dynamic sign-in page content render logic. 

## Linear Issue Reference

log-2164

## Testing

test locally 
ut case added
imageimage

imageimage

@logto-io/eng

### CODE REVIEW

File: packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx [hunk 1]
```diff
@@ -36,19 +38,28 @@ describe('SecondarySocialSignIn', () => {
...
   it('less than four connectors', () => {
-    const { container } = render(
-      <MemoryRouter>
-        <SecondarySocialSignIn connectors={socialConnectors.slice(0, 3)} />
-      </MemoryRouter>
+    const { container } = renderWithPageContext(
+      <SettingsProvider
+        settings={{
+          ...mockSignInExperienceSettings,
+          socialConnectors: socialConnectors.slice(0, 3),
```
Comment (demonzoo) [inline packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx:45]: nit: I've seen `socialConnectors.length > 3` and `socialConnectors.slice(0, 3)` numerous times. Can we somehow define a global variable for the `3` and use it everywhere?
Or maybe we can put this logic in the `useSocial` hook? Any thoughts?

File: packages/ui/src/hooks/use-social.ts [hunk 2]
```diff
@@ -64,11 +68,24 @@ const isNativeWebview = () => {
...
 
+  // Filter native supported social connectors
+  const socialConnectors = useMemo(
+    () =>
+      (experienceSettings?.socialConnectors ?? []).filter(({ id }) => {
+        if (!isNativeWebview()) {
+          return true;
+        }
+
+        return getLogtoNativeSdk()?.supportedSocialConnectors.includes(id);
```
Comment (demonzoo) [inline packages/ui/src/hooks/use-social.ts:84]: nit: maybe we can
```typescript
return !isNativeWebView() || getLogtoNativeSdk()?.supportedSocialConnectors.includes(id);
```

**Review by demonzoo — APPROVED** [review]
Overall looks good

File: packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx:45]: make sense let me check

Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx]: Update. Add it at the component level as it controls the UI display logic only and may differ between different components.

File: packages/ui/src/hooks/use-social.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/hooks/use-social.ts]: updated

**Pull request merged**